components/constellation/constellation.rs

```diff
@@ -750,6 +771,98 @@ where
         namespace_id
     }
 
+    fn get_event_loop(
+        &mut self,
+        host: &Host,
+        top_level_browsing_context_id: &TopLevelBrowsingContextId,
+        opener: &Option<BrowsingContextId>,
+    ) -> Option<Weak<EventLoop>> {
+        let event_loop = match opener {
+            Some(browsing_context_id) => {
+                let opener = self
+                    .browsing_contexts
+                    .get(&browsing_context_id)
+                    .expect("Opener BC to exist");
+                let bc_group_id = self
+                    .browsing_context_group_set
+                    .iter()
+                    .filter_map(|(id, bc_group)| {
+                        if bc_group
+                            .top_level_browsing_context_set
+                            .contains(&opener.top_level_id)
+                        {
+                            Some(id.clone())
+                        } else {
+                            None
+                        }
+                    })
+                    .last()
+                    .expect("The opener to have an existing BC group");
```

Comment:
Panic. Also, would it be easier to store the bc group id in the browsing context?

Reply:
Yes it appears much easier indeed